lib/solveEquation/stepThrough.js

```diff
@@ -50,9 +51,34 @@ function stepThrough(leftNode, rightNode, comparator, debug=false) {
   const MAX_STEP_COUNT = 20;
   let iters = 0;
 
+  // Checks if there are roots in the original equation before we
+  // do any simplification.
+  // E.g. (33 + 89) (x - 99) = 0
+  if (checks.canFindRoots(equation)) {
+    const [solutions, symbol] = getSolutionsAndSymbol(equation);
+
+    let allSolutions;
+
+    if (solutions.length > 1) {
+      const flatten = [];
+      solutions.forEach(s => s.items ? flatten.push(...s.items) : flatten.push(s));
+      allSolutions =  Node.Creator.list(flatten);
+    }
+    else {
+      allSolutions = solutions[0];
```

Comment:
When will this case occur? There is guaranteed a solution because there is a check (`canFindRoots`).